skimage/segmentation/_seeds.pyx

```diff
@@ -0,0 +1,725 @@
+    The resulting image size will be of shape `(metrics.h, metrics.w)`.
+    The size of the blocks that are scaled up is
+    `(metrics.block_h, metrics.block_w)`.
+    The number of blocks is `(metrics.n_blocks_y, metrics.n_blocks_x)`.
+    The last row and column are repeated if extra padding is required to make
+    up to `(metrics.h, metrics.w)`.
+
+    Parameters
+    ----------
+    labels_2d : the labels image to upscale
+    metrics : the `LevelMetrics` instance that describes the scale factor
+
+    Returns
+    -------
+    Upscaled label image of shape `(metrics.h, metrics.w)`
+    """
+    cdef int[:,:] up_labels_2d = np.zeros((metrics.h, metrics.w), dtype=np.int32)
+    cdef int x, y, i, j, block_w, block_h
+
+    for y in range(metrics.n_blocks_y):
+        if y < (metrics.n_blocks_y-1):
+            block_h = metrics.block_h
+        else:
+            block_h = metrics.last_block_h
+        for x in range(metrics.n_blocks_x):
+            if x < (metrics.n_blocks_x-1):
+                block_w = metrics.block_w
+            else:
+                block_w = metrics.last_block_w
+            for j in range(block_h):
+                for i in range(block_w):
+                    up_labels_2d[y*metrics.block_h+j,
+                                 x*metrics.block_w+i] = labels_2d[y,x]
+
+    return up_labels_2d
+
+
+@cython.boundscheck(False)
+def build_seed_histogram(int[:,:] index_2d, int hist_size, LevelMetrics metrics):
+    """
+    Build the base level histogram from the the provided index image.
+    Generates a histogram for each block of shape
+    `(metrics.block_h, metrics.block_w)`, of which the image is divided into
+    `(metrics.n_blocks_y, metrics.n_blocks_x)` blocks. Last rows and columns
+    of pixels from the image are included in their nearest respective blocks
+    if the image size does not divide exactly.
+
+    Parameters
+    ----------
+    index_2d : an index image giving the index of each pixel in the source
+    image
+    hist_size : histogram size (number of 'bars')
+    metrics : the `LevelMetrics` instance that describes the scale factor
+
+    Returns
+    -------
+    The histogram as an array of shape
+    `(metrics.n_blocks_y, metrics.n_blocks_x, hist_size)`
+    """
+    cdef int[:,:,:] hist_2d = np.zeros((metrics.n_blocks_y, metrics.n_blocks_x,
+                                        hist_size), dtype=np.int32)
+    cdef int x, y, i, j, block_h, block_w, n
+    for y in range(metrics.n_blocks_y):
+        if y < (metrics.n_blocks_y-1):
+            block_h = metrics.block_h
+        else:
+            block_h = metrics.last_block_h
+        for x in range(metrics.n_blocks_x):
+            if x < (metrics.n_blocks_x-1):
+                block_w = metrics.block_w
+            else:
+                block_w = metrics.last_block_w
+            for j in range(block_h):
+                for i in range(block_w):
+                    n = index_2d[y*metrics.block_h+j,x*metrics.block_w+i]
```

Comment:
You probably mean to use just `block_h` and `block_w` rather than `metrics.block_?` here? If so, please add a test that will catch this error in the future.